packages/flutter/lib/src/widgets/focus_traversal.dart

```diff
@@ -682,160 +903,575 @@ class _SortData {
+/// using an [OrderedTraversalPolicy].
+///
+/// By default, traverses in reading order using [ReadingOrderTraversalPolicy].
 ///
 /// See also:
 ///
 ///  * [FocusNode], for a description of the focus system.
-///  * [WidgetOrderFocusTraversalPolicy], a policy that relies on the widget
+///  * [WidgetOrderTraversalPolicy], a policy that relies on the widget
 ///    creation order to describe the order of traversal.
 ///  * [ReadingOrderTraversalPolicy], a policy that describes the order as the
 ///    natural "reading order" for the current [Directionality].
 ///  * [DirectionalFocusTraversalPolicyMixin] a mixin class that implements
 ///    focus traversal in a direction.
+class FocusTraversalGroup extends StatelessWidget {
+  /// Creates a [FocusTraversalGroup] object.
+  ///
+  /// The [child] argument must not be null.
+  FocusTraversalGroup({
+    Key key,
+    FocusTraversalPolicy policy,
+    @required this.child,
+    String debugLabel,
+  })  : _focusNode = FocusNode(
+          canRequestFocus: false,
+          skipTraversal: true,
+          debugLabel: debugLabel ?? 'FocusTraversalGroup',
+        ),
+        policy = policy ?? ReadingOrderTraversalPolicy(),
+        super(key: key);
+
+  /// The child widget of this [FocusTraversalGroup].
+  ///
+  /// {@macro flutter.widgets.child}
+  final Widget child;
+
+  // The internal focus node used to collect the children of this node into a
+  // group, and to provide a context for the traversal algorithm to sort the
+  // group with.
+  final FocusNode _focusNode;
+
+  /// The policy used to move the focus from one focus node to another when
+  /// traversing them using a keyboard.
+  ///
+  /// If not specified, traverses in reading order using
+  /// [ReadingOrderTraversalPolicy].
+  ///
+  /// See also:
+  ///
+  ///  * [FocusTraversalPolicy] for the API used to impose traversal order
+  ///    policy.
+  ///  * [WidgetOrderTraversalPolicy] for a traversal policy that traverses
+  ///    nodes in the order they are added to the widget tree.
+  ///  * [ReadingOrderTraversalPolicy] for a traversal policy that traverses
+  ///    nodes in the reading order defined in the widget tree, and then top to
+  ///    bottom.
+  final FocusTraversalPolicy policy;
+
+  /// Returns the focus policy group widget used by the [FocusTraversalPolicy]
+  /// that most tightly encloses the given [BuildContext].
+  ///
+  /// It does not create a rebuild dependency because changing the traversal
+  /// order doesn't change the widget tree, so nothing needs to be rebuilt as a
+  /// result of an order change.
+  ///
+  /// The `context` argument may only be null if `nullOk` is true.
+  ///
+  /// Will assert if no [FocusTraversalGroup] ancestor is found, and `nullOk` is false.
+  ///
+  /// If `nullOk` is true, then either having a null `context`, or not finding
+  /// an ancestor will return null.
+  static FocusTraversalGroup of(BuildContext context, {bool nullOk = false}) {
```

Comment:
assert context always non-null unless there's a good reason, in which case doc it :-)